### ISSUE TITLE
nontrivial SimpleLoopUnswitch crashes on partial unswitch assertion 

### ISSUE THREAD

**Issue by yrouban** [llvm/llvm-project#58666]
I found an IR that causes SimpleLoopUnswitch to crash as follows:
```console
$ opt -passes=simple-loop-unswitch'<nontrivial>' simple-loop-unswitch-crash.ll --disable-output
opt: /home/yrouban/work/llvm-project/llvm/lib/Transforms/Scalar/SimpleLoopUnswitch.cpp:2113: void unswitchNontrivialInvariants(llvm::Loop&, llvm::Instruction&,
llvm::ArrayRef<llvm::Value*>, llvm::IVConditionInfo&, llvm::DominatorTree&, llvm::LoopInfo&, llvm::AssumptionCache&, llvm::function_ref<void(bool, bool, llvm::A
rrayRef<llvm::Loop*>)>, llvm::ScalarEvolution*, llvm::MemorySSAUpdater*, llvm::function_ref<void(llvm::Loop&, llvm::StringRef)>): Assertion `((match(Cond, m_Log
icalAnd()) ^ match(Cond, m_LogicalOr())) || PartiallyInvariant) && "Only `or`, `and`, an `select`, partially invariant instructions " "can combine invariants be
ing unswitched."' failed.
```

[simple-loop-unswitch-crash.ll.txt](https://github.com/llvm/llvm-project/files/9888544/simple-loop-unswitch-crash.ll.txt)

**Comment by d-makogon** [comment]
Fix on review - https://reviews.llvm.org/D137249

**Comment by d-makogon** [comment]
Fixed by [ebac599](https://github.com/llvm/llvm-project/commit/ebac59999f926339f08936c050890c3364b5f130).

**Comment by fhahn** [comment]
Thanks for the fix!
